File: Serenity Practice/HosbitalSerenity.Web/Modules/Hosbital/MedicalRecords/MedicalRecordsColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace HosbitalSerenity.Hosbital.Columns;

[ColumnsScript("Hosbital.MedicalRecords")]
[BasedOnRow(typeof(MedicalRecordsRow), CheckNames = true)]
public class MedicalRecordsColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int RecordId { get; set; }
    public string DoctorName { get; set; }
    public string PatientName { get; set; }
    public DateTime RecordDate { get; set; }
    [EditLink]
    public string Notes { get; set; }
}

File: Serenity Practice/HosbitalSerenity.Web/Modules/ServerTypes/Hosbital/MedicalRecordsColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { MedicalRecordsRow } from "./MedicalRecordsRow";

export interface MedicalRecordsColumns {
    RecordId: Column<MedicalRecordsRow>;
    DoctorName: Column<MedicalRecordsRow>;
    PatientName: Column<MedicalRecordsRow>;
    RecordDate: Column<MedicalRecordsRow>;
    Notes: Column<MedicalRecordsRow>;
}

export class MedicalRecordsColumns extends ColumnsBase<MedicalRecordsRow> {
    static readonly columnsKey = 'Hosbital.MedicalRecords';
    static readonly Fields = fieldsProxy<MedicalRecordsColumns>();
}